src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -130,7 +129,8 @@ public Mono<Tuple2<Mono<TransactionActivatedEvent>, String>> handle(
                                                                                                 .getEcommerceFiscalCode(),
                                                                                         nodoOperations
                                                                                                 .generateRandomStringToIdempotencyKey()
-                                                                                )
+                                                                                ),
+                                                                                new ArrayList<>(5)
```

Comment:
why 5? better a constant with a meaningful name, also you need to update https://github.com/pagopa/pagopa-ecommerce-payment-requests-service/blob/main/src/main/kotlin/it/pagopa/ecommerce/payment/requests/repositories/PaymentRequestInfo.kt

Reply:
This is the max size of transferListItem. Since we know this value and it is quite small, we can set it to the max value, to get optimal performance from array list, as show here https://medium.com/@malith.jayasinghe/a-performance-evaluation-of-java-arraylist-f097582b5c4d . In particular this is the conclusion _"It is clear from the from the results (considering the add operation of ArrayList) that if the required maximum capacity of the ArrayList is known, we can get the optimal performance (both average latency and throughput) by specifying the initial capacity to final size final size (or using ensureCapacity as required)"_ 
To avoid memory waste, we can do an optimization like setting the initial capacity to 3 items or 1 till the cart won't be in exercise. When the cart will be real, we can tune this parameter. What do you think?

Speaking about payment-request I'm not so sure that we must do this editing in that side, since, before activation, that list will be null or empty. What do you think about this too?

Reply:
ok I agree for the optimizations, but why are you talking about a cart? `cart` and `transfertList` are decoupled

Reply:
You're right, I delete confusing consideration about cart. I was wrong.